src/stages/sqs-lambda.ts

```diff
@@ -116,6 +119,12 @@ export class SqsToLambdaStage extends DataStage {
       detailType: [eventDetailType],
     };
 
-    this.targets = [new events_targets.SqsQueue(this.queue)];
+    if (this.queue.fifo && props.messageGroupId == 'undefined') {
```

Comment:
You should do `!props.messageGroupId` instead. This will cover `undefined`, `null`, and empty strings.